### ISSUE TITLE
Samsung Remote Client: disconnected from *ip*

### ISSUE THREAD

**Issue by samuel-larsson** [CONNCTED/homebridge-samsungtv#2]
I can't get it to work at all. My TV's IP is correct in **config.json**. When I try to turn it on, nothing happens. When I try to turn it off (wether it's on or off already), I get:
`Samsung Remote Client: disconnected from 192.168.1.125:55000` 
in the console.

Any ideas?

**Comment by Irish-link** [comment]
Same issue here. I did a port scan and found that 55000 is in fact open along with a good amount of others. My first use i told the TV to allow the control of it so i'm not sure what might have happened.

**Comment by PilotConway** [comment]
I have the same issue here.

I looked in the settings on the TV and permission is set to allow control. I've also tried a few other apps in the app store and they all work without issue so I know I can remotely control the TV. 

The other apps do mention that you can only use remote control to turn off the TV, you can't turn it on, but even still homebridge isn't working to turn it off.

**Comment by johnschult** [comment]
Same problem here.

**Comment by stelfrich** [comment]
Got the same problem :+1:

**Comment by szerglyiphone** [comment]
I have a same problem any update about that?

**Comment by Irish-link** [comment]
So... For me what seemed to be happening is that it was doing this because of my style of tv. The other issue https://github.com/CONNCTED/homebridge-samsungtv/issues/1 seemeds to have fixed it for me. The original reason i didn't try it was because it seemed to be a seperate issue. This fixed it for me and i was able to turn off the tv with this fix.

**Comment by craigdent** [comment]
Any update on how to fix the Samsung remote client: disconnected issue

**Comment by szerglyiphone** [comment]
Hey Guys!

I have a 4K Samsung TV, HU6900 and I see the port is different. I check this, and I have 52345 port for allshare not 55000, anyway still not working. You can change your port a ./usr/local/lib/node_modules/homebridge-samsungtv/node_modules/samsung-remote/lib :)

```
15500/tcp open  unknown
52345/tcp open  http     Samsung AllShare httpd
```

I tired the KEY_POWEROFF too but nothing... :/ [#1](https://github.com/CONNCTED/homebridge-samsungtv/issues/1)

**Comment by davidbutler62** [comment]
I have the same problem.  I think the issue is with the Content Sharing settings on the TV itself..

Under Allshare settings > content settings I see:

NodeJS Samsung Remote 127.0.0.1 Allowed

I believe this is the problem.  127.0.0.1 should be replaced with the IP on my Raspberry Pi homebridge box.

Is there anyway to edit the address in the TV?  The only option i can see is to Allow, Deny or Remove

**Comment by craigdent** [comment]
david,
did you try this solution and did it work?

**Comment by davidbutler62** [comment]
no, because there is no way to change the allowed IP address in the Samsung TV setup menu.

It seems that the homebridge samsung TV plugin provides the loopback address instead of it's real ip address to the samsung tv the first time it connects

**Comment by craigdent** [comment]
so once 1 device connects to the Samsung it cant be changed or edited?
even if clicked disallow cant allow it later?

**Comment by D1ag0nall3y** [comment]
Not to reopen, but it hasn't been resolved. I am also experiencing this issue. Wondering if there is any plans on future development/updates to resolve this issue or if this is something TV specific and cannot be fixed with an update. Any input would be great.

**Comment by SRv6d** [comment]
@Danimal4326 Same here. I get  'Samsung Remote Client: disconnected from ...' Im using a UE48HU7500 Samsung 4K TV.

**Comment by gewoondemax** [comment]
Well here's the same problem, i tried to change it in Json and no work! what is happening? however All share sees the 127.0.0.1 NodeJS ... so there should be communication right? what about the Samsung Remote, this node is automatically installed with i installed version @0.0.5 ....

**Comment by socialsti** [comment]
Did anyone find a fix yet? getting the same issue.

**Comment by Danimal4326** [comment]
I don't have access to a newer TV, so there's no way for me to debug it.  I'm not sure if the underlying 'samsung-remote' module works on newer tv's

**Comment by socialsti** [comment]
Thanks, the issue seems to be that the IP is not being broadcasted correctly. It does connect the first time, but then doesn't work.

**Comment by ryck** [comment]
Same here, with a KS8000. Apparently, the protocol is not the same anymore (ie, the port is no longer 55000), and is probably encrypted now. I guess we will need to wait until someone cracks it :(

**Comment by Astroman** [comment]
Same with me on my KS8000. However, I also have the problem on my C8000 so I don't think it is only due to a change in protocol.
Edit: Implementing the solution to issue 4 allowed me to turn the power off on my C8000 but not my KS8000 so it may indeed be a protocol issue.

**Comment by apwelsh** [comment]
I had the issue, and looked at the source.  The node script only implements one command, POWER.  Most models require POWEROFF or POWERON (poweron does not typically work, since the TV disconnect network when it sleeps).  I wonder is the dev is open to my making the necessary fixes.

open the .../node_modules/homebridge-samsungtv/index.js
and change KEY_POWER to KEY_POWEROFF.

The volume and channel changing keys are not implemented, they are currently just wrappers for future support.

**Comment by Astroman** [comment]
I don't know if this helps, but the iOS app myTifi is able control my UN65KS8000, including turning it on. Therefore, it is indeed possible. :)

**Comment by apwelsh** [comment]
Yes, it is possible.  Generally, only works for those TVs that are connected via ethernet (not wifi), and that power off does not put ethernet to sleep.  It has been confirmed that there are many models that this works on.  And now we have another to add to the list.

I actually forked this project, modified the source to support both on/off commands as well as return off as a staus instead of an error when determing if tv is off.  After pushing my changes to github, I even created a pull request, for the dev to merge my changes, should he desire.  If you want to try my changes, my account on github is apwelsh

**Comment by ryck** [comment]
Yeah, the official app also works, so we know it's possible, but apparently they changed the protocol, and that's why this code is not working anymore

**Comment by apwelsh** [comment]
The official version behaves for me as reported in this thread -- it does nothing but report a disconnect. I have a UN55ES7150 which does have the prior generation GUI.  The only way to know if the protocol changed, or this this is simply a command error, is to either try my version, or sniff the network amd see what is there.  There mere fact the TV asks to accept the remote, suggests the protocol at issue in the report is the correct one.

**Comment by nunofgs** [comment]
> Yeah, the official app also works, so we know it's possible, but apparently they changed the protocol, and that's why this code is not working anymore

Is anyone working on dumping the new protocol?

**Comment by apwelsh** [comment]
I have a UN55ES7150.  This plugin did not work for me, so i ran a packet sniffer (WireShark) on the official app.  I found that the power command being used doesnt work, and so I changed the plugin to use the PWROFF command instead of the PWR command.  Now it works just fine for me.

I tried addressing this and was told that I am wrong and that it uses a new protocol.  I suggest you try using the doscreet power commands.

**Comment by ryck** [comment]
@apwelsh looks like your UN55ES7150 is not a 2016 model, so is still using the old protocol (well, or some version of it), and that's why is working for you at some level...
For the new TVs (2016 models, like KS8000) it doesn't work at all, because the protocol has changed

**Comment by kyleaa** [comment]
the 2016 protocol appears to use websockets with JSON. I was able to get this working on my 2016 model (UN60KS8000). published https://github.com/kyleaa/homebridge-samsungtv2016/blob/master/index.js but YMMV.

I connected with ethernet and implemented power on with wake-on-lan, siri is able to turn off and on my TV now. I presume if you connected with wifi only power off would work. Tested the `KEY_POWERON`/`KEY_POWEROFF` combo, they were not supported. `KEY_POWER` is the only one it will accept.

**Comment by hogendoorn** [comment]
Any update on this ?

**Comment by morphix9** [comment]
As it seems there is no fix for that. Are you guys using another plugin for Samsung?
